storage/store/delta.go

```diff
@@ -21,6 +22,12 @@ func (b *baseStore) ApplyDelta(delta *pbsubstreams.StoreDelta) {
 	keySize := uint64(len(delta.Key))
 	switch delta.Operation {
 	case pbsubstreams.StoreDelta_UPDATE:
+		for k := range b.recentlyDeletedPrefixes {
+			if strings.HasPrefix(delta.Key, k) {
+				delete(b.recentlyDeletedPrefixes, k) // we added a key matching this prefix, it is not considered deleted anymore
+			}
+		}
+
```

Comment:
This is repeated many times, it feels like maybe you should have a struct `DeletedPrefixesCache` or something like that and a single method would clean.